tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -9001,53 +9001,135 @@ TEST_F(ExecutionTest, BarycentricsTest) {
+              }
+            };
+
+        std::shared_ptr<ShaderOpTestResult> test2 = RunShaderOpTestAfterParse(pDevice, m_support, "BarycentricsCheckOrder", ResourceCallbackFn, ShaderOpSet);
+        MappedData data;
+        D3D12_RESOURCE_DESC &D = test2->ShaderOp->GetResourceByName("RTarget")->Desc;
+        UINT width = (UINT)D.Width;
+        UINT height = D.Height;
+        UINT pixelSize = GetByteSizeForFormat(D.Format);
+        test2->Test->GetReadBackData("RTarget", &data);
+        //const uint8_t *pPixels = (uint8_t *)data.data();
+        const float *pPixels = (float *)data.data();
+        // Get the vertex of barycentric coordinate using VBuffer
+        MappedData triangleData;
+        test2->Test->GetReadBackData("VBuffer", &triangleData);
+        const float *pTriangleData = (float*)triangleData.data();
+
+        // get the size of the input data
+        unsigned triangleVertexSizeInFloat = 0;
+        for (auto element : test2->ShaderOp->InputElements)
+            triangleVertexSizeInFloat += GetByteSizeForFormat(element.Format) / 4;
+        XMFLOAT2 p0(pTriangleData[0], pTriangleData[1]);
+        XMFLOAT2 p1(pTriangleData[triangleVertexSizeInFloat], pTriangleData[triangleVertexSizeInFloat + 1]);
+        XMFLOAT2 p2(pTriangleData[triangleVertexSizeInFloat * 2], pTriangleData[triangleVertexSizeInFloat * 2 + 1]);
+        XMFLOAT3 barycentricWeights[4] = {
+            XMFLOAT3(0.3333f, 0.3333f, 0.3333f),
+            XMFLOAT3(0.5f, 0.25f, 0.25f),
+            XMFLOAT3(0.25f, 0.5f, 0.25f),
+            XMFLOAT3(0.25f, 0.25f, 0.50f)
+        };
+        float tolerance = 0.001f;
+        for (unsigned i = 0; i < sizeof(barycentricWeights) / sizeof(XMFLOAT3); ++i) {
+            float w0 = barycentricWeights[i].x;
+            float w1 = barycentricWeights[i].y;
+            float w2 = barycentricWeights[i].z;
+            float x1 = w0 * p0.x + w1 * p1.x + w2 * p2.x;
+            float y1 = w0 * p0.y + w1 * p1.y + w2 * p2.y;
+            // map from x1 y1 to rtv pixels
+            int pixelX = (int)((x1 + 1) * (width - 1) / 2);
+            int pixelY = (int)((1 - y1) * (height - 1) / 2);
+            int offset = pixelSize * (pixelX + pixelY * width) / sizeof(pPixels[0]);
+            LogCommentFmt(L"location  %u %u, value %f, %f, %f, %f", pixelX, pixelY, pPixels[offset], pPixels[offset + 1], pPixels[offset + 2], pPixels[offset + 3]);
+          
+            // If the ordering constraint is met, then this pixel's RGBA should be all 1.0's
+            // since the shader only returns float4<1.0,1.0,1.0,1.0> when this condition is met.
+            VERIFY_IS_TRUE(CompareFloatEpsilon(pPixels[offset]    , 1.0, tolerance));
+            VERIFY_IS_TRUE(CompareFloatEpsilon(pPixels[offset + 1], 1.0, tolerance));
+            VERIFY_IS_TRUE(CompareFloatEpsilon(pPixels[offset + 2], 1.0, tolerance));
+            VERIFY_IS_TRUE(CompareFloatEpsilon(pPixels[offset + 3], 1.0, tolerance));
+        }
+
```

Comment:
A lot of this is duplicated with the above. Making a helper function would make it easier to compare to the old code and also allow any updates to be in just one place.